docs/Python-API.md

```diff
@@ -266,3 +268,172 @@ Once a property has been modified in Python, you can access it in C# after the n
 var sharedProperties = Academy.Instance.FloatProperties;
 float property1 = sharedProperties.GetPropertyWithDefault("parameter_1", 0.0f);
 ```
+
+#### [Advanced] Create your own SideChannel
+
+you can create your own `SideChannel` in C# and Python and use it to communicate data between the two.
+
+##### Unity side
+The side channel will have to implement the `SideChannel` abstract class. There are two methods
+that must be implemented :
+
+ * `ChannelType()` : Must return an integer identifying the side channel (This number must be the same on C#
+ and Python). There can only be one side channel of a certain type during communication.
+ * `OnMessageReceived(byte[] data)` : You must implement this method to specify what the side channel will be doing
+ with the data received from Python. The data is a `byte[]` argument.
+
+To send a byte array from C# to Python, call the `base.QueueMessageToSend(data)` method inside the side channel.
+The `data` argument must be a `byte[]`.
+
+To register a side channel on the Unity side, call `Academy.Instance.RegisterSideChannel` with the side channel
+as only argument.
+
+##### Python side
+The side channel will have to implement the `SideChannel` abstract class. You must implement :
+
+ * `channel_type(self) -> int` (property) : Must return an integer identifying the side channel (This number must
+be the same on C# and Python). There can only be one side channel of a certain type during communication.
+ * `on_message_received(self, data: bytes) -> None` : You must implement this method to specify what the
+ side channel will be doing with the data received from Unity. The data is a `byte[]` argument.
+
+To send a byte array from Python to C#, call the `super().queue_message_to_send(bytes_data)` method inside the
+side channel. The `bytes_data` argument must be a `bytes` object.
+
+To register a side channel on the Python side, call pass the side channel as argument when creating the
```

Comment:
This means that custom Side Channels and `mlagents-learn` are currently mutually exclusive.

@harperj Another use case for plugins.

Reply:
Yes, you are right. But I do not see how our training process would make use of custom made side channels. I see the side channels more as a research use case.

Reply:
Yeah, can't help but think some form of extension / plugin could allow us to add these more easily.  That said, I think you're right @vincentpierre that it isn't obvious how we could use these side channels even if the user could add them.